prodigy_demo/basic_recipe/text_classification_recipe_QUESTIONS.py

```diff
@@ -0,0 +1,139 @@
+"""
+textcat recipe with parameters to experiment with.
+
+We are experimenting with the following aspects of the recipe:
+
+1. Using a different baseline model;
+2. Filtering data to label and;
+3. Sorting examples based on model predictions.
+
+To run this recipe:
+
+prodigy textcat_hf tweets_annotated \
+    ./data/data_to_label_500.jsonl \
+    -F text_classification_recipe_QUESTIONS.py
+    
+"""
+from typing import List, Optional, Iterator
+import copy
+
+import prodigy
+from prodigy.components.loaders import JSONL
+from prodigy.components.preprocess import add_tokens
+import spacy
+
+# 1. USE A DIFFERENT BASELINE MODEL
+
+# Here, we're using the spacy-huggingface integration to load
+# a sentiment analysis transformers model from HuggingFace.
+
+# Can you experiment with different models that are appropriate for the task?
+# For example, you might want to use a transformers model fine tuned on tweets
+# 
+# i.e. finiteautomata/bertweet-base-sentiment-analysis. What's changed with the labels 
+# when you re-run the instance?
+
+#alternatively, you can simply use a spaCy model. 
+
+# here, we're loading a blank model
+nlp = spacy.blank("en")
+
+# now we're adding the huggingface model to the pipeline
+nlp.add_pipe(
+    "hf_text_pipe",
+    config={"model": "finiteautomata/bertweet-base-sentiment-analysis"},
+)
+
+# 2. FILTER DATA TO LABEL
+
+# Here, we have a filtering function to exlude examples
+# based on some filtering criteria.
+
+# Can you update this function to filter based on different criteria?
+# Perhaps its based on key words or POS tags.
+
+# As an extension, look into prodgy's pattern matcher
+# to filter based on patterns.
```

Comment:
any link you might want to provide?

Reply:
note to self to link out to:
1. On using pattern files in Prodigy: https://support.prodi.gy/t/how-to-use-a-spacy-pattern-in-prodigy/1561
2. On pattern matching in spaCy more generally: https://spacy.io/usage/rule-based-matching#adding-patterns-attributes